lib/spack/spack/test/llnl/util/filesystem.py

```diff
@@ -866,3 +866,18 @@ def test_visit_directory_tree_follow_none(noncyclical_dir_structure):
         j('b'),
     ]
     assert not visitor.symlinked_dirs_after
+
+
+@pytest.mark.regression('29687')
+@pytest.mark.parametrize('initial_mode', [
+    stat.S_IRUSR | stat.S_IXUSR,
+    stat.S_IWGRP
+])
+@pytest.mark.skipif(sys.platform == 'win32', reason='Windows might change permissions')
+def test_remove_linked_tree_doesnt_change_file_permission(tmpdir, initial_mode):
```

Comment:
Can you add a docstring here that explains the scenario that the test is checking? Other than that all the changes here look great.

Reply:
Done in 07780d7